modules/standard/Socket.chpl

```diff
@@ -0,0 +1,1403 @@
+  the maximum number of allowed pending connections in the system.
+*/
+var backlogDefault:uint(16) = (if SOMAXCONN <= 128 then SOMAXCONN else 128):uint(16);
+
+/*
+  Convenience procedure which creates a new :type:`tcpListener` bound
+  to and listening on `address` for new connections. `backlog`
+  determines how many connections can be pending (not having called
+  accept) before the socket will begin to reject them. The default
+  value of backlog is `backlogDefault`.
+
+  .. code-block:: Chapel
+
+    const address = ipAddr.create("127.0.0.1", 8000, IPFamily.IPv4);
+    const server = listen(address);
+
+  :arg address: address to connect to
+  :type address: :type:`ipAddr`
+  :arg timeout: standard ipv6 address.
+  :type timeval: :type:`~Sys.timeval`
+  :return: connected socket.
+  :rtype: `tcpConn`
+  :throws SystemError: On failure to bind or listen on `address`
+*/
+proc listen(in address: ipAddr, reuseAddr: bool = true, backlog: uint(16) = backlogDefault): tcpListener throws {
+  var family = address.family;
+  var socketFd = socket(family, SOCK_STREAM);
+  bind(socketFd, address, reuseAddr);
+  var err_out = sys_listen(socketFd, backlog:c_int);
+  if err_out != 0 {
+    throw SystemError.fromSyserr(err_out, "Failed to listen on socket");
+  }
+  const listener = new tcpListener(socketFd);
+  return listener;
+}
+
+/*
+  Convenience procedure which creates a :type:`tcpConn` connected to
+  `address`.`timeout` determines how much time to wait for
+  connection to be established. The default value for `timeout` is
+  indefinite.
+
+  .. code-block:: Chapel
+
+    const address = ipAddr.create("127.0.0.1", 8000, IPFamily.IPv4);
+    const timeout = new timeval(4,0);
+    const connectedClient = connect(address, timeout);
+
+  :arg address: address to connect to
+  :type address: :type:`ipAddr`
+  :arg timeout: time to wait for connection establishment.
+  :type timeval: :type:`~Sys.timeval`
+  :return: connected socket.
+  :rtype: `tcpConn`
+  :throws SystemError: Upon failure to connect.
+*/
+proc connect(const ref address: ipAddr, in timeout = new timeval(-1,0)): tcpConn throws {
+  var family = address.family;
+  var socketFd = socket(family, SOCK_STREAM);
+  setBlocking(socketFd, false);
+  var err_out = sys_connect(socketFd, address._addressStorage);
+  if err_out != 0 && err_out != EINPROGRESS {
+    sys_close(socketFd);
+    throw SystemError.fromSyserr(err_out,"connect() failed");
+  }
+  if err_out == 0 {
+    setBlocking(socketFd, true);
+    return openfd(socketFd):tcpConn;
+  }
+  var localSync$: sync int = 0;
+  localSync$.readFE();
+  var writerEvent = event_new(event_loop_base, socketFd, EV_WRITE | EV_TIMEOUT, c_ptrTo(syncRWTCallback), c_ptrTo(localSync$):c_void_ptr);
+  defer {
+    event_del(writerEvent);
+    event_free(writerEvent);
+  }
+  if timeout.tv_sec == -1 {
+    err_out = event_add(writerEvent, nil);
+  }
+  else {
+    err_out = event_add(writerEvent, c_ptrTo(timeout));
+  }
```

Comment:
It could help throwing some empty lines between logical parts of the code. But nonetheless, this if/else is just

```chapel
err_out = event_add(writerEvent,
                    if timeout.tv_sec==-1 then nil else c_ptrTo(timeout));
```